website/src/audio-player.tsx

```diff
@@ -45,201 +54,93 @@ interface Props {
   slices?: { start: number; end: number }
   existingHowl?: Howl
 }
-class AudioPlayer extends React.Component<
-  Props,
-  { progress: number; howl: Howl }
-> {
-  constructor(props: Props) {
-    super(props)
-    // let slices = this.props.slices && {
-    //   ...this.props.slices.map((a) => {this.props.slices.indexOf(a)}), ...this.props.slices
-    // }
-    this.state = {
-      progress: 0,
-      howl: new Howl({
-        src: [this.props.audioUrl],
-        html5: true,
-        format: [".wav"],
-        preload: this.props.preload || false,
-        onplay: () => {
-          this.onPlay()
-        }, // What happens when a sound starts playing?
-        onend: () => {}, // What happens when a sound finishes playing?
-        onpause: () => {}, // What happens when a sound is paused?
-        onseek: () => {}, // What happens when a sound seeks?
-      }),
-    }
-    if (this.props.slices) {
-      let slices = {
-        __default: [
-          this.props.slices.start,
-          this.props.slices.end - this.props.slices.start,
-        ],
-        sound: [
-          this.props.slices.start,
-          this.props.slices.end - this.props.slices.start,
-        ],
-      }
-      Object.assign(this.state.howl, { _sprite: slices })
-    }
-    // Event handlers
-    this.nextStep = this.nextStep.bind(this)
-    this.onPlay = this.onPlay.bind(this)
-    this.onSeek = this.onSeek.bind(this)
-  }
 
-  onPlay() {
-    requestAnimationFrame(this.nextStep)
-  }
+const AudioComponent = (props: Props) => {
+  const audio = useRef(new Audio(props.audioUrl))
 
-  nextStep() {
-    let seek = this.state.howl.seek() || 0
-    this.setState((prevState, props) => ({
-      progress: (seek / this.state.howl.duration()) * 100,
-    }))
-    if (this.state.howl.playing()) {
-      requestAnimationFrame(this.nextStep)
+  const [progress, setProgress] = useState(0)
+  const [loadStatus, setLoadStatus] = useState(0)
+
+  useEffect(() => {
+    const interval = setInterval(() => {
+      setProgress(audio.current.currentTime)
+      setLoadStatus(audio.current.readyState)
```

Comment:
There's no listeners for load progress at all on the Audio api?

Reply:
There are some event listeners and loading can be rearranged on that. There is also an event listener for the current time, but I think to avoid that since using that onEvent winds up being much choppier than checking current time every set time period.